tests/testthat/test-strings_as_factors_linter.R

```diff
@@ -0,0 +1,148 @@
+test_that("strings_as_factors_linter skips allowed usages", {
+  expect_lint("data.frame(1:3)", NULL, strings_as_factors_linter())
+  expect_lint("data.frame(x = 1:3)", NULL, strings_as_factors_linter())
+
+  expect_lint("data.frame(x = 'a', stringsAsFactors = TRUE)", NULL, strings_as_factors_linter())
+  expect_lint("data.frame(x = 'a', stringsAsFactors = FALSE)", NULL, strings_as_factors_linter())
+  expect_lint("data.frame(x = c('a', 'b'), stringsAsFactors = FALSE)", NULL, strings_as_factors_linter())
+
+  # strings in argument names to c() don't get linted
+  expect_lint("data.frame(x = c('a b' = 1L, 'b c' = 2L), stringsAsFactors = FALSE)", NULL, strings_as_factors_linter())
```

Comment:
Why provide `stringsAsFactors` in this test case?

Reply:
Hmm maybe just sloppy-pasted. But turns out there's a real false positive there!

Probably not picked up on our side because quoted keyword args aren't allowed if not needed, and passing literally-named vectors to `data.frame()` seems uncommon. So let's file as a follow-up bug for now.